util/codec/codec.go

```diff
@@ -176,7 +176,7 @@ func EncodeMySQLTime(sc *stmtctx.StatementContext, t types.Time, tp byte, b []by
 	if tp == mysql.TypeUnspecified {
 		tp = t.Type
 	}
-	if tp == mysql.TypeTimestamp && sc.TimeZone != time.UTC {
+	if tp == mysql.TypeTimestamp && sc != nil && sc.TimeZone != time.UTC {
```

Comment:
sc can be nil?

Reply:
in one of previous commit, sc maybe nil, but in current it should never be nil, remove this change.